src/pages/Post.jsx

```diff
@@ -0,0 +1,20 @@
+import Header from 'components/Header';
+import { Helmet } from 'react-helmet';
+import { Outlet } from 'react-router-dom';
+import { useMediaQuery } from 'react-responsive';
+
+function Post() {
+  const isMobile = useMediaQuery({ minWidth: 767 });
+  return (
+    <>
+      <Helmet>
+        <title>Rolling - post</title>
+      </Helmet>
+      {isMobile && <Header />}
+      <Outlet />
+    </>
+  );
```

Comment:
헤더 조건부 렌더링과 관련된 상태관리를 상위 컴포넌트로 옮겨주신것 좋아요 👍👍
모바일이면서 post페이지일 때, 헤더를 보여줄지 말지 여부를 헤더 내부에서 결정하지 않고 상위 컴포넌트인 Main으로 관련된 로직을 옮기면 main과 post 레이아웃 컴포넌트를 분리하지 않아도 되고, 조건부 렌더링 중복도 피할 수 있을 것 같아요
```suggestion
  const [isHeaderVisible, setIsHeaderVisible] = useState(true);
  const location = useLocation();
  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isPostPage = location.pathname.startsWith('/post');
  
  useEffect(() => {
    const handleHeaderVisibility = () => {
      const isHeaderHidden = isMobile && isPostPage;
      setIsHeaderVisible(!isHeaderHidden);
    };

    handleHeaderVisibility();

    window.addEventListener('resize', handleHeaderVisibility);

    return () => {
      window.removeEventListener('resize', handleHeaderVisibility);
    };
  }, [isMobile, isPostPage]);
  
  return (
    <>
      <Helmet>
        <title>Rolling - main </title>
      </Helmet>
      {isHeaderVisible && <Header />}
      <Outlet />
    </>
  );
```

Reply:
이걸 생각못했넵 수정할게용 ㅎㅎ